packages/router/src/router_state.ts

```diff
@@ -248,9 +248,9 @@ function flattenInherited(pathFromRoot: ActivatedRouteSnapshot[]): Inherited {
   return pathFromRoot.reduce((res, curr) => {
     const params = {...res.params, ...curr.params};
     const data = {...res.data, ...curr.data};
-    const resolve = {...res.resolve, ...curr._resolvedData};
+    const resolve = {...res.resolve, ...curr.routeConfig?.data, ...curr._resolvedData};
```

Comment:
```suggestion
    const resolve = {...curr.data, ...res.resolve, ...curr.routeConfig?.data, ...curr._resolvedData};
```

Some developers have apparently discovered they can directly mutate the data in their guards/resolvers. We'll need to also copy `curr.data` to prevent that from breaking.

Reply:
@atscott Should I include a test for this use-case even when it's probably not a recommended way of using it.

Reply:
Yea, probably. You can add a comment to the test case as well to explain why it's there. i.e. "This is not documented or recommended behavior but is here to prevent unexpected regressions. This behavior isn't necessary 'by design' but it was discovered during a refactor that some teams depend on it."

Reply:
Actually, it took me a while to figure out what scenario would not work without `...curr.data` :).